### PR TITLE
Feature - KhuyenDang - TabBar

### PR DESCRIPTION
**Pull request by mvn-khuyendang-dn**

### CODE REVIEW

File: 13_TabbarController/KhuyenDang/TabBar/TabBar/NavigationController/LoginVIewController/LoginViewController.swift [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            break
+        }
+        let scene = UIApplication.shared.connectedScenes.first
+        if let sceneDelegate : SceneDelegate = (scene?.delegate as? SceneDelegate) {
+            sceneDelegate.changeScreen(type: .tabbar)
+        }
+    }
+}
+
+extension LoginViewController: UITextFieldDelegate {
```
Comment (mvn-tanhuynh-dn) [inline 13_TabbarController/KhuyenDang/TabBar/TabBar/NavigationController/LoginVIewController/LoginViewController.swift:97]: // MARK: - UITextFieldDelegate

File: 13_TabbarController/KhuyenDang/TabBar/TabBar/NavigationController/LoginVIewController/LoginViewController.swift [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        if textField == usernameTextField {
+            passwordTextField.becomeFirstResponder()
+        }
+        else if textField == passwordTextField {
+            handleLogin()
+        }
```
Comment (mvn-tanhuynh-dn) [inline 13_TabbarController/KhuyenDang/TabBar/TabBar/NavigationController/LoginVIewController/LoginViewController.swift:104]: if ... {
...
} else if ... {
...
} else {
...
}

File: 13_TabbarController/KhuyenDang/TabBar/TabBar/NavigationController/RegisterViewController/RegisterViewController.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        let doneButton = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(handleRegister))
+        navigationItem.rightBarButtonItem = doneButton
+    }
+    
+    @objc private func handleRegister() {
+        // code handle register
+        let scene = UIApplication.shared.connectedScenes.first
+        if let sceneDelegate : SceneDelegate = (scene?.delegate as? SceneDelegate) {
+            sceneDelegate.changeScreen(type: .tabbar)
+        }
```
Comment (mvn-tanhuynh-dn) [inline 13_TabbarController/KhuyenDang/TabBar/TabBar/NavigationController/RegisterViewController/RegisterViewController.swift:28]: Create BaseViewController, all ViewControllers will inherit.
Define change root function to share.

File: 13_TabbarController/KhuyenDang/TabBar/TabBar/TabBarController/ProfileViewController/ProfileViewController.swift [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        super.viewDidLoad()
+        title = "Profile"
+    }
+
+    // MARK: - IBAction
+    @IBAction private func logOutButtonTouchUpInside(_ sender: Any) {
+        let scene = UIApplication.shared.connectedScenes.first
+        if let sceneDelegate : SceneDelegate = (scene?.delegate as? SceneDelegate) {
+            sceneDelegate.changeScreen(type: .login)
+        }
```
Comment (mvn-tanhuynh-dn) [inline 13_TabbarController/KhuyenDang/TabBar/TabBar/TabBarController/ProfileViewController/ProfileViewController.swift:22]: Create BaseViewController, all ViewControllers will inherit.
Define change root function to share.

File: 13_TabbarController/KhuyenDang/TabBar/TabBar/NavigationController/LoginVIewController/LoginViewController.swift [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+            guard let value = value as? NSDictionary else{ return }
+            guard let username = value["username"] as? String, usernameTextField.text == username,
+                  let password = value["password"] as? String, passwordTextField.text == password else { return }
+            let homeViewController = HomeViewController()
+            navigationController?.pushViewController(homeViewController, animated: true)
+            break
+        }
+        let scene = UIApplication.shared.connectedScenes.first
+        if let sceneDelegate : SceneDelegate = (scene?.delegate as? SceneDelegate) {
+            sceneDelegate.changeScreen(type: .tabbar)
```
Comment (mvn-tanhuynh-dn) [inline 13_TabbarController/KhuyenDang/TabBar/TabBar/NavigationController/LoginVIewController/LoginViewController.swift:92]: Create BaseViewController, all ViewControllers will inherit.
Define change root function to share.

**Comment by mvn-tamnguyen7-dn** [comment]
GLTM!

**Pull request merged**